mlir/utils/performance/perfRunner.py

```diff
@@ -414,28 +429,39 @@ def fromCommandLine(cls, argv, arch):
             else:
                 continue
 
-        return cls(dataType, direction, layout, n, c, hi, wi, k, y, x,
+        if layout is None:
+            raise ValueError(f"Missing layouts")
+        if (filterLayout is None):
+            filterLayout = layout
+        if (inputLayout is None):
+            inputLayout = layout
+        if (outputLayout is None):
+            outputLayout = layout
+        return cls(dataType, direction, filterLayout, inputLayout, outputLayout, n, c, hi, wi, k, y, x,
             convStrideH, convStrideW, paddingH, paddingW, dilationH, dilationW,
                    group, arch)
 
-    def __init__(self, dtype: str, direction: str, layout: str,
+    def __init__(self, dtype: str, direction: str, fLayout: str, iLayout:str, oLayout:str,
                     n: int, c: int, hi: int, wi: int, k: int, y: int, x: int,
                     convStrideH: int, convStrideW: int, paddingH: int, paddingW: int,
                     dilationH: int, dilationW: int, group: int, arch: str):
         if dtype not in {"f16", "f32", "bf16", "i8"}:
             raise ValueError(f"Invalid datatype: {dtype}")
         if direction not in {"fwd", "bwd", "wrw"}:
             raise ValueError(f"Invalid direction: {direction}")
-        if layout not in self.MLIR_OUTPUT_LAYOUTS:
-            raise ValueError(f"Invalid layout: {layout}")
+        #if layout not in self.MLIR_OUTPUT_LAYOUTS:
+        #    raise ValueError(f"Invalid layout: {layout}")
 
         self.MLIR_N_REPEATS = 5
         self.dataType = dtype
         self.direction = direction
 
-        self.filterLayout = self.MLIR_FILTER_LAYOUTS[layout]
-        self.inputLayout = layout.lower()
-        self.outputLayout = self.MLIR_OUTPUT_LAYOUTS[layout]
+        #self.filterLayout = self.MLIR_FILTER_LAYOUTS[layout]
+        #self.inputLayout = layout.lower()
+        #self.outputLayout = self.MLIR_OUTPUT_LAYOUTS[layout]
+        self.filterLayout = ''.join(self.filterLayoutMap.get(c).lower() for c in fLayout)
+        self.inputLayout = ''.join(self.inputLayoutMap.get(c).lower() for c in iLayout)
```

Comment:
Given the context, this should probably be `.get(c, c)` so that MLIR-style layouts (like `ngkhw`) don't throw exceptions.